samples/simpleasset/fungibleasset.go

```diff
@@ -0,0 +1,266 @@
+	}
+
+	id := getWalletId(owner)
+	return addFungibleAssetsHelper(ctx, id, fungibleAssetType, numUnits)
+}
+
+// DeleteFungibleAssets burns the fungible assets from an owner.
+func (s *SmartContract) DeleteFungibleAssets(ctx contractapi.TransactionContextInterface, fungibleAssetType string, numUnits int, owner string) error {
+	exists, err := s.FungibleAssetTypeExists(ctx, fungibleAssetType)
+	if err != nil {
+		return err
+	}
+	if !exists {
+		return fmt.Errorf("the fungible asset type %s does not exist.", fungibleAssetType)
+	}
+
+	id := getWalletId(owner)
+	return subFungibleAssetsHelper(ctx, id, fungibleAssetType, numUnits)
+}
+
+// TransferFungibleAssets transfers the fungible assets from an owner to newOwner
+func (s *SmartContract) TransferFungibleAssets(ctx contractapi.TransactionContextInterface, fungibleAssetType string, numUnits int, owner string, newOwner string) error {
+	exists, err := s.FungibleAssetTypeExists(ctx, fungibleAssetType)
+	if err != nil {
+		return err
+	}
+	if !exists {
+		return fmt.Errorf("the fungible asset type %s does not exist.", fungibleAssetType)
+	}
+
+
+	ownerId := getWalletId(owner)
+	newOwnerId := getWalletId(newOwner)
+
+	err = subFungibleAssetsHelper(ctx, ownerId, fungibleAssetType, numUnits)
+	if err != nil {
+		return err
+	}
+	err = addFungibleAssetsHelper(ctx, newOwnerId, fungibleAssetType, numUnits)
+	if err != nil {
+		// Revert subtraction from the original owner
+		// Assuming following will succeed (not sure what to do if it does not)
+		_ = addFungibleAssetsHelper(ctx, ownerId, fungibleAssetType, numUnits)
+		return err
+	}
+	return nil
+}
+
+// GetBalance returns the amount of given fungible asset type owned by an owner.
+func (s *SmartContract) GetBalance(ctx contractapi.TransactionContextInterface, fungibleAssetType string, owner string) (int, error) {
+	exists, err := s.FungibleAssetTypeExists(ctx, fungibleAssetType)
+	if err != nil {
+		return -1, err
+	}
+	if !exists {
+		return -1, fmt.Errorf("the fungible asset type %s does not exist.", fungibleAssetType)
+	}
+
+	id := getWalletId(owner)
+	walletJSON, err := ctx.GetStub().GetState(id)
+	if err != nil {
+		return -1, fmt.Errorf("failed to read owner's wallet from world state: %v", err)
+	}
+
+	var wallet Wallet
+	err = json.Unmarshal(walletJSON, &wallet)
+	if err != nil {
+		return -1, err
+	}
+	balance := wallet.WalletMap[fungibleAssetType]
+	return balance, nil
+}
+
+// Checks if owner has some given amount of fungible asset
+func (s *SmartContract) FungibleAssetsExist(ctx contractapi.TransactionContextInterface, fungibleAssetType string, numUnits int, owner string) (bool, error) {
+	balance, err := s.GetBalance(ctx, fungibleAssetType, owner)
+	if err != nil {
+		return false, err
+	}
+	return balance >= numUnits, nil
+}
+
+func addFungibleAssetsHelper(ctx contractapi.TransactionContextInterface, id string, fungibleAssetType string, numUnits int) error {
```

Comment:
Capitalize `add`.

Reply:
That is a helper function, not a member function of Smart Contract. Should I still capitalize that?